Zend/zend_execute.c

```diff
@@ -1985,12 +2119,29 @@ static zend_always_inline void zend_fetch_property_address(zval *result, zval *c
 	    EXPECTED(Z_OBJCE_P(container) == CACHED_PTR_EX(cache_slot))) {
 		uint32_t prop_offset = (uint32_t)(intptr_t)CACHED_PTR_EX(cache_slot + 1);
 		zend_object *zobj = Z_OBJ_P(container);
-		zval *retval;
 
 		if (EXPECTED(prop_offset != (uint32_t)ZEND_DYNAMIC_PROPERTY_OFFSET)) {
-			retval = OBJ_PROP(zobj, prop_offset);
-			if (EXPECTED(Z_TYPE_P(retval) != IS_UNDEF)) {
-				ZVAL_INDIRECT(result, retval);
+			ptr = OBJ_PROP(zobj, prop_offset);
+			if (EXPECTED(Z_TYPE_P(ptr) != IS_UNDEF)) {
+return_indirect:
+				ZVAL_INDIRECT(result, ptr);
+				if ((by_ref & ZEND_FETCH_REF)
+				 && (prop_op_type == IS_CONST
+				  || EXPECTED(Z_TYPE_P(prop_ptr) == IS_STRING))) {
```

Comment:
This check is weird and redundant. line 2118 anyway required prop_op_type == IS_CONST.